atlasdb-refactorings/src/test/java/com/palantir/atlasdb/refactorings/UseTypedUserAgentVsStringTest.java

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.refactorings;
+
+import org.junit.Test;
+
+import com.palantir.baseline.refaster.RefasterTestHelper;
+
+public class UseTypedUserAgentVsStringTest {
+
+    @Test
+    public void canConvertFromToStringUserAgent() {
+        RefasterTestHelper.forRefactoring(UseTypedUserAgentVsString.class)
+                .withInputLines("Test",
+                        "import com.codahale.metrics.MetricRegistry;",
+                        "import com.palantir.atlasdb.factory.TransactionManagers;",
+                        "import com.palantir.atlasdb.keyvalue.api.KeyValueService;",
+                        "import com.palantir.conjure.java.api.config.service.UserAgent;",
+                        "import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;",
+                        "",
+                        "class Test {",
+                        "    public void main(String[] args) {",
+                        "        UserAgent userAgent = UserAgent.of(UserAgent.Agent.of(\"product\", \"1.2.3\"));",
+                        "        KeyValueService kvs = TransactionManagers.builder()",
+                        "                .config(null)",
+                        "                .userAgent(userAgent.toString())",
+                        "                .globalMetricsRegistry(new MetricRegistry())",
+                        "                .globalTaggedMetricRegistry(DefaultTaggedMetricRegistry.getDefault())",
+                        "                .build()",
+                        "                .serializable()",
+                        "                .getKeyValueService();",
+                        "        System.out.println(kvs.isInitialized());",
+                        "    }",
+                        "}")
+                .hasOutputLines(
+                        "import com.codahale.metrics.MetricRegistry;",
+                        "import com.palantir.atlasdb.factory.TransactionManagers;",
+                        "import com.palantir.atlasdb.keyvalue.api.KeyValueService;",
+                        "import com.palantir.conjure.java.api.config.service.UserAgent;",
+                        "import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;",
+                        "",
+                        "class Test {",
+                        "    public void main(String[] args) {",
+                        "        UserAgent userAgent = UserAgent.of(UserAgent.Agent.of(\"product\", \"1.2.3\"));",
+                        "        KeyValueService kvs = TransactionManagers.builder()",
+                        "                .config(null).userAgent(userAgent)",
```

Comment:
Why does it replace the newline?

Reply:
no idea, couldn't get it to work otherwise 🤷‍♂ will ask infra about it

Reply:
nope - this is why we're investing in a formatter.